### PR TITLE
refactor(gui-client): reload log filter immediately

### PR DESCRIPTION
**Pull request by ReactorScram**
This will simplify #5590 some. The API URL and auth URL still take effect on the next sign-in, but we don't have to explain that the settings take effect after restarting the entire Client process, those take effect somewhat immediately.

For some reason I see some lag, maybe the tracing layers don't check for a new filter on every span, maybe they have some delay to save CPU time.

### CODE REVIEW

File: rust/gui-client/src-tauri/src/client/gui.rs [hunk 1]
```diff
@@ -489,11 +492,15 @@ impl Controller {
...
                 self.advanced_settings = settings;
-                // TODO: Update the logger here if we can. I can't remember if there
-                // was a reason why the reloading didn't work.
-                tracing::info!(
-                    "Applied new settings. Log level will take effect at next app start."
+                self.log_filter_reloader
+                    .reload(filter)
+                    .context("Couldn't reload log filter")?;
+                tracing::debug!(
+                    "Applied new settings. Log level will take effect immediately for the GUI and later for the IPC service."
```
Comment (thomaseizinger) [inline rust/gui-client/src-tauri/src/client/gui.rs:503]: Can we send an IPC message and reload the filter there too?

**Review by thomaseizinger — APPROVED** [review]

File: rust/gui-client/src-tauri/src/client/gui.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/gui-client/src-tauri/src/client/gui.rs:503]: Yeah that's the plan. Right now the IPC connection is only up when we're signed in, so I'm going to change that so the GUI connects immediately when it starts.

Comment (ReactorScram) [reply rust/gui-client/src-tauri/src/client/gui.rs:503]: That will also unblock #5453

Comment (thomaseizinger) [reply rust/gui-client/src-tauri/src/client/gui.rs:503]: Sounds great :)

**Pull request merged**